pymedphys/_dicom/delivery/core.py

```diff
@@ -161,20 +163,28 @@ def _from_dicom_beam(cls, beam, meterset):
         )
 
         dicom_mlcs = rtplan.get_leaf_jaw_positions_for_type(
-            beam_limiting_device_position_sequences, "MLCX"
+            beam_limiting_device_position_sequences, "MLC"
         )
 
-        mlcs = [
-            np.array(
-                [-np.array(mlc[0:num_leaves][::-1]), np.array(mlc[num_leaves::][::-1])][
-                    ::-1
-                ]
-            ).T
-            for mlc in dicom_mlcs
-        ]
+        mlcs = np.array(
+            [
+                np.array(
+                    [
+                        -np.array(mlc[0:num_leaves][::-1]),
+                        np.array(mlc[num_leaves::][::-1]),
+                    ][::-1]
+                ).T
+                for mlc in dicom_mlcs
+            ]
+        )
+
+        # Nasty hack, patching the fact that the leaf banks are swapped
```

Comment:
this is high risk.
where is the unit test for this?

Reply:
Yup, I agree. I need to make a test based upon the following data:

[unity-trf-and-dicom.zip](https://github.com/pymedphys/pymedphys/files/5474408/unity-trf-and-dicom.zip)

Reply:
I'd hoped to hear back with some results from Martijn's testing, and potentially some simple DICOM data and TRF data that would be ideal for testing with.

See email above:

> Keen for you to put it through its paces and report back. Also, if there are a few simple RT plan DICOM files with short deliveries and their corresponding TRFs over a range of cases I would highly appreciate being able to include that data within the pymedphys testing suite.

Unfortunately haven't heard back yet. My "Delivery Object" coordinate system uses as its reference the TRF file format. So in this case, essentially the key is that it is verified that the delivery object representation created from the DICOM file is equivalent to that produced by the TRF file.

Reply:
That's all good and fine. I wouldn't have it any other way. I was planning tomorrow at work to build a test and mark it slow based on the data above. If we get more refined data I can speed up the test by pulling something in.

Reply:
Any further thoughts on the following PR?

https://github.com/pymedphys/pymedphys/pull/1111